app/controllers/katello/api/v2/host_collections_controller.rb

```diff
@@ -1,10 +1,17 @@
 module Katello
   class Api::V2::HostCollectionsController < Api::V2::ApiController
     include Katello::Concerns::FilteredAutoCompleteSearch
-    before_action :find_host_collection, :only => [:copy, :show, :update, :destroy,
-                                                   :add_hosts, :remove_hosts, :hosts]
-    before_action :find_activation_key
-    before_action :find_host
+    before_action :find_authorized_katello_resource, :only => [
+      :copy,
+      :show,
+      :update,
+      :destroy,
+      :add_hosts,
+      :remove_hosts,
+      :hosts
+    ]
+    before_action :find_readable_activation_key
```

Comment:
I see this method only being used by index method.
```suggestion
    before_action :find_readable_activation_key, :only => [:index]
```